### DISCUSSION TITLE
Need to draw gratings of different contrast and angle

### DISCUSSION THREAD

**Discussion by Mirudhula-m** [jspsych/jsPsych#1119] [Q&A]
Hi,

I need to draw some gratings in a circular placeholder, and these gratings have to be generated at certain instances of time during the experiment, and each of these instances may have a different kind of grating i.e. different angles and contrast. 
I understand that I can make use of SVGs to build it from the scratch. But, these will have to be generated dynamically during the course of the experiment, and therefore will take time to generate, in addition to extra time it will take to render on the screen. This would, then, affect the stimulus display timings.

I would like to know if there is an easier way to generate and render these gratings dynamically without affecting stimulus timings.

Any help or suggestion would be appreciated. Thank you!

**Comment by becky-gilbert** [comment]
I'm wondering if this is something that could be done with the help of @kurokida's psychophysics plugin? Here's the [github repo](https://github.com/kurokida/jspsych-psychophysics) and [website](http://jspsychophysics.hes.kyushu-u.ac.jp/).

**Reply by Mirudhula-m** [reply]
This looks really promising! I will try it out and let you know soon. Thank you!

**Reply by kurokida** [reply]
Thank you for considering using my plugin.

Perhaps [demos/manual_drawFunc_gradation.html](https://www.hes.kyushu-u.ac.jp/~kurokid/jspsychophysics/demos/manual_drawFunc_gradation.html) is close to what you are looking for.
However, it may be difficult to render the gratings quickly even though using my plugin. I think you'd be better off using webgl, but I don't implement this in my plugin.

**Reply by Mirudhula-m** [reply]
Thank you for your plugin and the suggestion! 
I will try implementing them soon, and post my experience here, so that anyone else can also benefit from it.
